mono/mini/mini.h

```diff
@@ -1538,11 +1540,13 @@ typedef enum {
 	MONO_CFG_HAS_TYPE_CHECK = 1 << 9
 } MonoCompileFlags;
 
+#ifdef MONO_ARCH_SIMD_INTRINSICS
```

Comment:
What ifdefing an enum out buys us?

Reply:
There is a possible rationale, of minimizing what you declare, to only what works. It helps people  avoid later errors. This is like producing slimmed down BCL metadata so calling nonexistant functions fails to autocomplete/compile instead throwing a runtime exception.

Personally I strive to minimize ifdefs and there are way too many in mono and the wider world.
In order to have fewer variations to compile and perhaps to link, so more functionality can be had from fewer binaries, unless performance suffers much.